src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.Optional;
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
+import mate.academy.hibernate.relations.lib.DataProcessingException;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
-    @Override
+    private MovieDao movieDao;
+    private SessionFactory sessionFactory;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+        movieDao = new MovieDaoImpl(sessionFactory);
+    }
+
     public Movie add(Movie movie) {
-        return null;
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+        Optional<Movie> movieOptional = movieDao.get(id);
+        if (movieOptional.isPresent()) {
+            return movieOptional.get();
+        }
+        throw new DataProcessingException("Not found in DB movie with id = " + id);
```

Comment:
dataProcessing is for dao layer
in service layer use entityNotFoundException
we should handle optional more gracefully - use orElseThrow here
fix everywhere